### ISSUE TITLE
Consider converting system channels to standard codec

### ISSUE THREAD

**Issue by stuartmorgan** [flutter/flutter#53188]
Currently many of the Flutter framework's system channels use the JSON codec, presumably purely for legacy reasons. This is not ideal for a number of reasons:
- Not all languages/platforms have a standard JSON library, so some embeddings (e.g., our Windows and Linux embeddings) need to use/link/ship a third-party JSON library just to talk to critical system channels.
  - In fact, the Flutter engine is shipping RapidJSON purely for this reason, I believe.
- It essentially forces every new platform to implement two codecs: standard for compatibility with most (~all? I don't know if we have data on this), and JSON for compatibility with the framework.
- It's less efficient than the standard method codec (and the general advantages of JSON over binary don't really apply to ephemeral, internal messages)

This would be a breaking change for all embedding implementations. Some options:
- Just make it a breaking change. It should be very easy for people to fix (since they will almost certainly have implemented the standard codec anyway)
- Add parallel channels with new names, and an opt-in method for an embeddings to say they want the new ones (set at engine startup).

/cc @Hixie for thoughts.

**Comment by blasten** [comment]
standard codec requires mapping of field names to [indexes in the buffer.](https://github.com/googleads/googleads-mobile-flutter/blob/master/packages/google_mobile_ads/android/src/main/java/io/flutter/plugins/googlemobileads/AdMessageCodec.java#L28)  

Maybe we could migrate more plugins to use Pigeon, which already addresses this issue. https://github.com/flutter/packages/blob/master/packages/pigeon/lib/dart_generator.dart#L66

**Comment by stuartmorgan** [comment]
> standard codec requires mapping of field names to [indexes in the buffer.](https://github.com/googleads/googleads-mobile-flutter/blob/master/packages/google_mobile_ads/android/src/main/java/io/flutter/plugins/googlemobileads/AdMessageCodec.java#L28)

No it doesn't, that's just more efficient. Almost every plugin I've looked at uses dictionaries for method arguments, not array indexes, and they are all using the standard codec.

**Comment by blasten** [comment]
That's right.  thank you. Pigeon uses a Map too.  I presume that implementation detail can be changed if it turns out to be critical.
